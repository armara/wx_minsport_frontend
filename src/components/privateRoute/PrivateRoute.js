import React from 'react'
import { Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'

import { isAuthenticated } from 'utils/auth'
import { getUserWorkplaceType } from 'utils/user'

const PrivateRoute = ({
  render,
  exact,
  path,
  location: { pathname = '' } = {},
}) => {
  return (
    <Route
      exact={exact}
      path={path}
      render={() => {
        if (!isAuthenticated()) {
          return <Redirect to="/login" />
        }

        if (pathname === '/') {
          return <Redirect to={`/${getUserWorkplaceType()}`} />
        }

        return render()
      }}
    />
  )
}

PrivateRoute.defaultProps = {
  exact: false,
  location: {},
}

PrivateRoute.propTypes = {
  render: PropTypes.func.isRequired,
  path: PropTypes.string.isRequired,
  location: PropTypes.shape({ pathname: PropTypes.string }),
  exact: PropTypes.bool,
}

export default PrivateRoute
