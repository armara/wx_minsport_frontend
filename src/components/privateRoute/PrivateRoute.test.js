import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'

import PrivateRoute from 'components/privateRoute/PrivateRoute'

configure({ adapter: new Adapter() })

describe('<PrivateRoute />', () => {
  it('renders correctly default props', () => {
    localStorage.setItem('user', '{"test":"ok"}')
    const tree = renderer
      .create(
        <Router>
          <PrivateRoute path="/" render={() => <div />} />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
