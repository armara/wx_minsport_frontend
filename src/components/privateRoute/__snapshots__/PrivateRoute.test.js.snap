// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> renders correctly default props 1`] = `<div />`;
